### ISSUE TITLE
Behaviour when input file is empty

### ISSUE THREAD

**Issue by multimeric** [shenwei356/csvtk#176]
### Prerequisites

- [x] make sure you're are using the latest version by `csvtk version`
- [x] read the [usage](http://bioinf.shenwei.me/csvtk/usage/)

### Describe your issue

* [x] describe the problem: we get `xopen: no content` when an empty file is piped into `csvtk` commands. In this case I would hope for the commands to work as though they are dealing with a valid 0-row table.
* [x] provide a reproducible example:
    ```bash
   echo '' | csvtk add-header --names "foo,bar"
    ```
   Here I would expect the output file to be:
   ```
   foo,bar
   ```
   However we get:
   ```
   [ERRO] xopen: no content
   ```

This isn't a super crucial feature, but it would be nice to have, so that pipelines don't break when the input files are empty, which is a fairly common edge case for me.

**Comment by shenwei356** [comment]
Yes, you're right, tolerance to empty files should be taken into consideration for most commands. Right now, only one command `dim` is implemented.

For now, I can only report the file name with no content, so users can check them.
[csvtk.tar.gz](https://github.com/shenwei356/csvtk/files/7703339/csvtk.tar.gz)

```
$ echo '' | csvtk add-header --names "foo,bar"
[ERRO] empty file: -
```

Since there are so many commands to modify, I'll work on it when I have enough time.

**Comment by shenwei356** [comment]
solved in #204
